src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -1226,7 +1226,11 @@ module private AstPass =
         | "removeAt" ->
             icall "splice" (c.Value, [args.Head; makeConst 1]) |> Some
         | "reverse" when kind = Array ->
-            icall "reverse" (instanceArgs c i.args) |> Some
+            match i.returnType with
+            | Fable.Array _ ->
+                emit i "$0.slice().reverse()" i.args |> Some
```

Comment:
Ok, I guess if you use ` emit i "$0.slice().reverse()" (instanceArgs c i.args) |> Some` instead it should work. That's equivalent to `defaultArg i.callee i.args.Head`.